service/src/main/java/tds/config/services/impl/ConfigServiceImpl.java

```diff
@@ -45,11 +45,17 @@ public ConfigServiceImpl(ConfigRepository configRepository,
     }
 
     @Override
-    public Optional<ClientSystemFlag> findClientSystemFlag(String clientName, String auditObject) {
+    public Optional<ClientSystemFlag> findClientSystemFlag(String clientName, String type) {
+        /*
+            The type is the name of the ClientSystemFlag.  Ideally the source of data for this
+            method will be an in-memory collection; ClientSystemFlag values rarely change, making them a
+            strong candidate for caching.
+        */
+
```

Comment:
I think we should be retrieving this data by client/type.  I think if I understand Spring's caching we could make those long term and they'd be fast.

If we choose to keep this data in memory without using the caching strategy a better structure would be a Map with the key being the clientname and type.